src/sage/numerical/pivot_rules_for_simplex_method.py

```diff
@@ -0,0 +1,776 @@
+        sage: D = P.initial_dictionary()
+        sage: pivot = SimplexMethodPivot("dantzig")
+        sage: pivot(D)
+        sage: D.entering()
+        x1
+        sage: D.leaving()
+        x4
+        """
+        if current_dictionary.entering() is None:
+            entering_index = argmax(current_dictionary.objective_coefficients())
+            entering_variable = current_dictionary.nonbasic_variables()[entering_index]
+            current_dictionary.enter(entering_variable)
+        if current_dictionary.leaving() is None:
+            possible = current_dictionary.possible_leaving()
+            if possible:
+                leaving_variable = min(current_dictionary.ratios())[1]
+                current_dictionary.leave(leaving_variable)
+
+class NWRule(AbstractSimplexMethodPivotRule):
+    r"""
+    Selects the entering and leaving variable based on a Normalzied Weight (NW) pivot rule.
+
+    For a linear program `min(cx: Ax\leq b) = min(cx: x\in P)`, an NW-rule is defined by a normalization `\eta: \mathbb{R}^d\to \mathbb{R}` and a weight `w\in \mathbb{R}^n`.
+    Given `v\in P` s.t. `cv \neq cv_{\text{opt}}`, pick the next vertex `u_* = \text{argmax}\{ \frac{w^t(u-v)}{\eta(u-v)}:` `u` adjacent to `v` and `c^tu>c^tv\}`.
+    See [BDLS2022]_.
+
+    EXAMPLES::
+
+        sage: from sage.numerical.pivot_rules_for_simplex_method import SimplexMethodPivot
+        sage: A = ([1, 1], [3, 1])
+        sage: b = (1000, 1500)
+        sage: c = (10, 5)
+        sage: P = InteractiveLPProblemStandardForm(A, b, c)
+        sage: D = P.initial_dictionary()
+        sage: pivot = SimplexMethodPivot("NW_rule")
+        sage: def eta(v): # use 2 norm as normalization
+        ....:     return v.norm()
+        sage: pivot(D, False, eta, [1,2])
+        sage: D.entering()
+        x2
+        sage: D.leaving()
+        x3
+
+    NW rule is not eqipped to solve both a Phase I and a phase II with :meth:`run_simplex_method`::
+
+        sage: A = ([1, 1], [3, 1], [-1, -1])
+        sage: b = (1000, 1500, -400)
+        sage: c = (10, 5)
+        sage: P = InteractiveLPProblemStandardForm(A, b, c)
+        sage: def eta(v):
+        ....:     return v.norm()
+        sage: P.run_simplex_method('NW_rule', eta, [1,2])
+        Traceback (most recent call last):
+        ...
+        TypeError: unsupported operand parent(s) for *: 'Ambient free module of rank 2 over the principal ideal domain Integer Ring' and 'Vector space of dimension 3 over Rational Field'
+
+    A Phase I and a Prase II problem require different NW rules due to the difference in ambient dimension of the respective problems::
+
+        sage: A = ([1, 1], [3, 1], [-1, -1])
+        sage: b = (1000, 1500, -400)
+        sage: c = (10, 5)
+        sage: P = InteractiveLPProblemStandardForm(A, b, c)
+        sage: def eta(v):
+        ....:     return v.norm()
+        sage: D = P.auxiliary_problem().initial_dictionary() # Phase I
```

Comment:
```suggestion
        sage: D = P.auxiliary_problem().initial_dictionary()  # Phase I
```
two spaces before `#` in inline comments (check throughout)